frontend/catalyst/jax_primitives.py

```diff
@@ -413,6 +415,63 @@ def _grad_lowering(ctx, *args, jaxpr, fn, grad_params):
     ).results
 
 
+# value_and_grad
+#
+@value_and_grad_p.def_impl
+def _value_and_grad_def_impl(ctx, *args, jaxpr, fn, grad_params):  # pragma: no cover
+    raise NotImplementedError()
+
+
+@value_and_grad_p.def_abstract_eval
+def _value_and_grad_abstract(*args, jaxpr, fn, grad_params):  # pylint: disable=unused-argument
+    """This function is called with abstract arguments for tracing.
+    Note: argument names must match these of `_value_and_grad_lowering`."""
+    return jaxpr.out_avals + jaxpr.out_avals
```

Comment:
Just for my curiousity (I am not too familiar with value and grad). Here JAXPR represents the function that will be derived and the out_avals are its outputs. Is the output always going to be the same types repeated? I remember when I implemented a little bit of gradients we had to adjust the signature based on which partial derivative we were taking. Maybe this is already taken into account and I just forgot.

Reply:
Yes great point, have a look at the abstract eval of grad, the outvals are correct here but the gradient types need to be deduced correctly.